mediaapi/routing/upload.go

```diff
@@ -158,6 +158,12 @@ func (r *uploadRequest) doUpload(
 		}
 	}
 
+	// If Content-Length header is not set, check if temp file size exceeds max file size configuration
+	if r.MediaMetadata.FileSizeBytes < 1 && bytesWritten > types.FileSizeBytes(*cfg.MaxFileSizeBytes) {
```

Comment:
Please universally do this check i.e:
```go
if bytesWritten > types.FileSizeBytes(*cfg.MaxFileSizeBytes) {
```
as we will want to assert this regardless of what the client puts as the Content-Length.